File: Historical/src/models/CoinTable.py
from sqlalchemy import Table, MetaData, Column, String, Double, DateTime, Integer

metaObj = MetaData()

CoinTableModel = Table(
    "CoinTable",
    metaObj,
    Column("Symbol", String, nullable=False, autoincrement=False, unique=False),
    Column("OpenDateTime", DateTime, nullable=False, autoincrement=False, unique=False),
    Column("CloseDateTime", DateTime, nullable=False, autoincrement=False, unique=False),
    Column("Open", Double, nullable=False, autoincrement=False, unique=False),
    Column("High", Double, nullable=False, autoincrement=False, unique=False),
    Column("Low", Double, nullable=False, autoincrement=False, unique=False),
    Column("Close", Double, nullable=False, autoincrement=False, unique=False),
    Column("Volume", Double, nullable=False, autoincrement=False, unique=False),
    Column("NumTrades", Integer, nullable=True, autoincrement=False, unique=False)
)

File: Daily/src/models/CoinTable.js
const { DataTypes, Model } = require('sequelize');
const { sequelize } = require('../database/SQLite');


class CoinTableModel extends Model {}

CoinTableModel.init({
    index: {
        type: DataTypes.INTEGER,
        primaryKey: true
    },
    Symbol: {
        type: DataTypes.STRING,
        allowNull: false,
        autoIncrement: false,
        unique: false
    },
    OpenDateTime: {
        type: DataTypes.DATE,
        allowNull: false,
        autoIncrement: false,
        unique: false
    },
    CloseDateTime: {
        type: DataTypes.DATE,
        allowNull: false,
        autoIncrement: false,
        unique: false
    },
    Open: {
        type: DataTypes.DOUBLE,
        allowNull: false,
        autoIncrement: false,
        unique: false
    },
    High: {
        type: DataTypes.DOUBLE,
        allowNull: false,
        autoIncrement: false,
        unique: false
    },
    Low: {
        type: DataTypes.DOUBLE,
        allowNull: false,
        autoIncrement: false,
        unique: false
    },
    Close: {
        type: DataTypes.DOUBLE,
        allowNull: false,
        autoIncrement: false,
        unique: false
    },
    Volume: {
        type: DataTypes.DOUBLE,
        allowNull: false,
        autoIncrement: false,
        unique: false
    },
    NumTrades: {
        type: DataTypes.INTEGER,
        allowNull: true,
        autoIncrement: false,
        unique: false,
        defaultValue: -1
    }
}, {sequelize, modelName: 'CoinTable', tableName:'CoinTable', timestamps: false});

module.exports = {CoinTableModel};
